### ISSUE TITLE
Backtracking error

### ISSUE THREAD

**Issue by VandaLovejoy** [ViennaRNA/ViennaRNA#93]
Hello,
I am trying to run my multiple alignment sequences with RNALalifold. I keep getting a backtracking error. 
This is the error I get:
![trial`](https://user-images.githubusercontent.com/44384386/94185393-3a3d2000-fe73-11ea-9f67-56146b6ed9eb.JPG)
When I remove the substrings indicated in the error message, the last 20 chars of the alignments, it works fine.
![image](https://user-images.githubusercontent.com/44384386/94185640-8d16d780-fe73-11ea-92b1-491213d178cc.png)
This is what the last 20 characters, the problematic ones look like 
![image](https://user-images.githubusercontent.com/44384386/94185825-cea78280-fe73-11ea-85a2-610c107d4ffb.png)
However, when I run RNALalifold only on the last 20 chars, no problem:
![image](https://user-images.githubusercontent.com/44384386/94186022-20500d00-fe74-11ea-85af-05efe1913cec.png). 
Anyone have any ideas why this is? The problematic sequences always seem to be at the end of the whole sequence.

**Comment by RaumZeit** [comment]
Hi,
is it possible for you to provide us with your input alignment that causes the backtracking failures with `RNALalifold`?
Otherwise, I fear we are not able to reproduce your problem.

**Comment by RaumZeit** [comment]
Hi again,
so it seems that I was able to identify the problem with the help of an other users bug report.
The actual cause of the problem is some undefined behavior when adding dangling end contributions for consensus structure predictions in cases where the alignment consists of more than 32768 columns. This would lead to a short integer overflow, and internally bogus numbers would be reported.
I've already isolated the problem throughout our library and am in the process to prepare a new release that fixes this error.

I'm very sorry for the inconveniences this problem has caused!

**Comment by VandaLovejoy** [comment]
Hi Ronny,
Sorry for taking so long to reply. I've been busy with writing scholarships. What you are saying would make sense. I fixed the problem when I add a 0 at the end of the alignments.
Vanda

On Oct. 9, 2020 4:51 a.m., Ronny Lorenz <notifications@github.com> wrote:

Hi again,
so it seems that I was able to identify the problem with the help of an other users bug report.
The actual cause of the problem is some undefined behavior when adding dangling end contributions for consensus structure predictions in cases where the alignment consists of more than 32768 columns. This would lead to a short integer overflow, and internally bogus numbers would be reported.
I've already isolated the problem throughout our library and am in the process to prepare a new release that fixes this error.

I'm very sorry for the inconveniences this problem has caused!

**Comment by VandaLovejoy** [comment]
Sorry I forgot to say, my bug happened at this line 172. I added a 0 because it was mentioned in the RNAlib here:
https://www.tbi.univie.ac.at/RNA/ViennaRNA/doc/html/group__aln__utils.html#ga7886fc678553c06472a712a335cde9cf
That alignments needed to be null terminated.
Thank you so much for spending time looking into this.
Vanda Gaonac'h-Lovejoy

On Oct. 9, 2020 4:51 a.m., Ronny Lorenz <notifications@github.com> wrote:

Hi again,
so it seems that I was able to identify the problem with the help of an other users bug report.
The actual cause of the problem is some undefined behavior when adding dangling end contributions for consensus structure predictions in cases where the alignment consists of more than 32768 columns. This would lead to a short integer overflow, and internally bogus numbers would be reported.
I've already isolated the problem throughout our library and am in the process to prepare a new release that fixes this error.

I'm very sorry for the inconveniences this problem has caused!
